### PR TITLE
Add support for filtering the schedules table

### PR DESCRIPTION
**Pull request by j-f1**
You can currently search by payee name, account name, next date, amount, and status (e.g. upcoming/missed).

### CODE REVIEW

**Comment by Kidglove57** [comment]
Very nice indeed. Thank you for responding so swiftly to my issue #589. From my wish list there have been some great usability additions this past week or so. So thank you all.

**Comment by MatissJanis** [comment]
Looks like there's a bug somewhere. The search isn't really changing the results in the table.

Screenshot 2023-02-02 at 18 26 29

**Comment by j-f1** [comment]
works fine for me on the deploy preview:


Screenshot_2023-02-02 13 28 26

**Comment by MatissJanis** [comment]
Try adding a schedule without account or payee (leave it blank). Those get ignored by the search query.

**Review by MatissJanis — COMMENTED** [review]
Overall looks good! Found no more bugs.

LMK what you think about the comment and then I'm happy to approve.

File: packages/desktop-client/src/components/schedules/SchedulesTable.js [hunk 1]
```diff
@@ -126,31 +128,70 @@ export function ScheduleAmountCell({ amount, op }) {
...
+  let accounts = useCachedAccounts();
+
+  let filteredSchedules = useMemo(() => {
+    if (!filter) {
+      return schedules;
+    }
+    const filterIncludes = str =>
+      str
+        ? str.toLowerCase().includes(filter.toLowerCase()) ||
+          str.toLowerCase().includes(filter.toLowerCase())
```
[the remark this answers is not in the document]
Comment (j-f1) [reply packages/desktop-client/src/components/schedules/SchedulesTable.js:155]: Oops! That’s supposed to match if either the query is a substring of the schedule text or the schedule text is a substring of the query.

**Comment by j-f1** [comment]
I also believe this is not susceptible to a variant of #614.

**Review by MatissJanis — APPROVED** [review]

**Comment by MatissJanis** [comment]
> I believe this PR is _not_ susceptible to a variant of #614.

Yup. I was not able to reproduce the bug in this PR for schedules.

**Pull request merged**